src/silverlabnwb/rois.py

```diff
@@ -0,0 +1,145 @@
+        for column in self.column_mapping:
+            if column not in self.type_mapping:
+                self.type_mapping[column] = np.float16
+        roi_data = pd.read_csv(
+            roi_path, sep='\t', header=0, index_col=False,
+            converters=self.type_mapping, memory_map=True)
+        # Rename the columns so that we can use them as identifiers later on
+        roi_data.rename(columns=self.column_mapping, inplace=True)
+        # Convert some columns as required
+        roi_data = roi_data.astype(self.type_conversion_post_read)
+        return roi_data
+
+    def get_row_attributes(self, roi_row):
+        return {
+            field: getattr(roi_row, field)
+            for field in self.column_mapping.values()
+        }
+
+
+class ClassicRoiReader(RoiReader):
+    """A reader for older versions of LabView setup (up to 2.1.3)."""
+    def __init__(self):
+        self.column_mapping = self.base_column_mapping
+        self.type_mapping = self.base_type_mapping
+        self.type_conversion_post_read = self.base_type_conversion_post_read
+
+
+class RoiReaderv300(RoiReader):
+    """A reader for LabView version 3.0.0."""
+    def __init__(self):
+        self.column_mapping = self.base_column_mapping
+        self.column_mapping.update({
+            'Resolution': 'resolution',
+            'Dwell Time per pixel': 'pixel_dwell_time',
+            'Pixels per miniscan': 'pixels_per_miniscan',
+            'Original FOV (um)': 'original_fov_um',
+            'Apparent Frame Size': 'apparent_frame_size'
+        })
+        self.type_mapping = self.base_type_mapping
+        # We don't need any more conversions while reading? But if we do,
+        # they can be added here.
+        self.type_conversion_post_read = self.base_type_conversion_post_read
+        self.type_conversion_post_read.update({
+            # For if we need to convert any of the new columns post-read.
+        })
+
+
+class RoiReaderv300Variable(RoiReaderv300):
+    """A reader for LabView version 3.0.0, supporting variable shape ROIs."""
+
+    def get_roi_imaging_plane(self, roi_number, plane_name, nwb_file):
+        """
+        Gets the imaging plane for a variable size ROI.
+
+        Overrides base method for variable size ROIs to create (if necessary, it may have been created previously
+        for a different optical channel) and return an appropriately spatially calibrated imaging plane. This is
+        needed because variable size ROIs need to have their own imaging plane, unlike fixed sized ROIs,
+        which can share an imaging plane with other ROIs. """
+        roi_row_idx = [index for index, value in enumerate(nwb_file.roi_data.roi_index) if value == roi_number]
+        assert len(roi_row_idx) == 1
+        roi_row_idx = roi_row_idx[0]
+        resolution = nwb_file.roi_data.resolution[roi_row_idx]
+        z_plane = nwb_file.nwb_file.processing['Acquired_ROIs'].get("ImageSegmentation")[plane_name].imaging_plane
```

Comment:
So we're still assuming here that each ROI falls entirely within an existing Z plane?